tests/integration/test_dbstatecheck.py

```diff
@@ -0,0 +1,132 @@
+import pytest
+from textwrap import dedent
+from typing import TYPE_CHECKING
+
+from cylc.flow.dbstatecheck import CylcWorkflowDBCheckerContext as Checker
+
+
+if TYPE_CHECKING:
+    from cylc.flow.dbstatecheck import CylcWorkflowDBChecker
+
+
+@pytest.fixture(scope='module')
+async def checker(
+    mod_flow, mod_scheduler, mod_run, mod_complete
+) -> 'CylcWorkflowDBChecker':
+    """Make a real world database.
+
+    We could just write the database manually but this is a better
+    test of the overall working of the function under test.
+    """
+    wid = mod_flow({
+        'scheduling': {
+            'graph': {'P1Y': dedent('''
+                good:succeeded
+                bad:failed?
+                output:custom_output
+            ''')},
+            'initial cycle point': '1000',
+            'final cycle point': '1001'
+        },
+        'runtime': {
+            'bad': {'simulation': {'fail cycle points': '1000'}},
+            'output': {'outputs': {'trigger': 'message'}}
+        }
+    })
+    schd = mod_scheduler(wid, paused_start=False)
+    async with mod_run(schd):
+        await mod_complete(schd)
+        schd.pool.force_trigger_tasks(['1000/good'], [2])
+        # Allow a cycle of the main loop to pass so that flow 2 can be
+        # added to db
+        await sleep(1)
+        with Checker(
+            'somestring', 'utterbunkum',
+            schd.workflow_db_mgr.pub_path
+        ) as checker:
+            yield checker
+
+
+def test_basic(checker):
+    """Pass no args, get unfiltered output"""
+    result = checker.workflow_state_query()
+    expect = [
+        ['bad', '10000101T0000Z', 'failed', '[1]'],
+        ['bad', '10010101T0000Z', 'succeeded', '[1]'],
+        ['good', '10000101T0000Z', 'succeeded', '[1]'],
+        ['good', '10010101T0000Z', 'succeeded', '[1]'],
+        ['output', '10000101T0000Z', 'succeeded', '[1]'],
+        ['output', '10010101T0000Z', 'succeeded', '[1]'],
+        ['good', '10000101T0000Z', 'waiting', '[2]'],
+        ['good', '10010101T0000Z', 'waiting', '[2]'],
+    ]
+    assert result == expect
+
+
+def test_task(checker):
+    """Filter by task name"""
+    result = checker.workflow_state_query(task='bad')
+    assert result == [
+        ['bad', '10000101T0000Z', 'failed', '[1]'],
+        ['bad', '10010101T0000Z', 'succeeded', '[1]']
+    ]
+
+
+def test_point(checker):
+    """Filter by point"""
+    result = checker.workflow_state_query(cycle='10000101T0000Z')
+    assert result == [
+        ['bad', '10000101T0000Z', 'failed', '[1]'],
+        ['good', '10000101T0000Z', 'succeeded', '[1]'],
+        ['output', '10000101T0000Z', 'succeeded', '[1]'],
+        ['good', '10000101T0000Z', 'waiting', '[2]'],
+    ]
+
+
+def test_status(checker):
+    """Filter by status"""
+    result = checker.workflow_state_query(status='failed')
+    expect = [
+        ['bad', '10000101T0000Z', 'failed', '[1]'],
+    ]
+    assert result == expect
+
+
+def test_output(checker):
```

Comment:
Indicates this  https://github.com/cylc/cylc-flow/pull/5809/files#r1586443251 breakage